tests/testthat/test-get_code.R

```diff
@@ -264,6 +251,40 @@ testthat::test_that("@linksto tag indicate affected object if object is assigned
   )
 })
 
+testthat::test_that("get_code works for assign detection no matter how many parametrers were provided in assign", {
+  code <- c(
+    "x <- 1",
+    "assign(\"x\", 0, envir = environment())",
+    "assign(inherits = FALSE, immediate = TRUE, \"z\", 5, envir = environment())",
+    "y <- x + z",
+    "y <- x"
+  )
+
+  tdata <- eval_code(teal_data(), code)
+
+  testthat::expect_identical(
+    get_code(tdata, datanames = "y"),
+    paste(code, collapse = "\n")
+  )
+
+})
+
+testthat::test_that("get_code detects function usage of assignment operator", {
+  code <- c(
+    "x <- 1",
+    "`<-`(y,x)"
```

Comment:
I'm realising that this will be endless struggle with code parsing. We can fill 99.9% of the gaps in `get_code` but there will be always something which can break the parser. Let's keep it experimental and sit down at some point to discuss whether code subsetting has that big value for `teal`.